src/components/SingleSelect/SingleSelect.jsx

```diff
@@ -0,0 +1,196 @@
+	},
+
+	getDefaultProps() {
+		return {
+			allowReset: true,
+			isDisabled: false,
+			selectedIndex: null,
+			DropMenu: DropMenu.getDefaultProps()
+		};
+	},
+
+	getInitialState() {
+		return {
+			optionGroups: [],
+			flattenedOptionsData: [],
+			ungroupedOptionData: [],
+			optionGroupDataLookup: {}
+		}
+	},
+
+	componentWillMount() {
+		// preprocess the options data before rendering
+		this.setState(DropMenu.preprocessOptionData(this.props, SingleSelect));
+	},
+
+	componentWillReceiveProps(nextProps) {
+		// only preprocess options data when it changes (via new props) - better performance than doing this each render
+		this.setState(DropMenu.preprocessOptionData(nextProps, SingleSelect));
+	},
+
+	render() {
+		const {
+			style,
+			className,
+			allowReset,
+			isDisabled,
+			selectedIndex,
+			onSelect,
+			DropMenu: dropMenuProps
+		} = this.props;
+
+		const {
+			direction,
+			isExpanded
+		} = dropMenuProps;
+
+		const {
+			optionGroups,
+			optionGroupDataLookup,
+			ungroupedOptionData,
+			flattenedOptionsData
+		} = this.state;
+
+		const placeholderProps = _.first(SingleSelect.Placeholder.findInAllAsProps(this.props));
+		const placeholder = _.get(placeholderProps, 'children', 'Select');
+		const isItemSelected = _.isNumber(selectedIndex);
+
+		return (
+			<DropMenu
+				{...dropMenuProps}
+				isDisabled={isDisabled}
+				selectedIndices={isItemSelected ? [selectedIndex] : []}
+				className={boundClassNames('&', className)}
+				onSelect={onSelect}
+				style={style}
+			>
+				<DropMenu.Control>
+					<div className={boundClassNames('&-Control', {
+						'&-Control-is-selected': !isDisabled && isItemSelected,
+						'&-Control-is-expanded': isExpanded,
+						'&-Control-is-disabled': isDisabled
+					})}>
+						<span
+							{...(!isItemSelected ? placeholderProps : null)}
+							className={boundClassNames(
+									'&-Control-content',
+									(!isItemSelected ? _.get(placeholderProps, 'className') : null)
+							)}
+						>
+							{isItemSelected ? flattenedOptionsData[selectedIndex].optionProps.children : placeholder}
+						</span>
+						<CaretIcon direction={isExpanded ? direction : 'down'} />
+					</div>
+				</DropMenu.Control>
+				{allowReset && isItemSelected ? (
+					<DropMenu.NullOption {...placeholderProps}>{placeholder}</DropMenu.NullOption>
+				) : null}
+				{
+					// for each option group passed in, render a DropMenu.OptionGroup, any label will be included in it's children, render each option inside the group
+					_.map(optionGroups, (optionGroupProps, optionGroupIndex) => {
+						return (
+							<DropMenu.OptionGroup key={'SingleSelectOptionGroup' + optionGroupIndex} {...optionGroupProps}>
+								{optionGroupProps.children}
+								{_.map(_.get(optionGroupDataLookup, optionGroupIndex), ({ optionProps, optionIndex }) => (
+									<DropMenu.Option key={'SingleSelectOption' + optionIndex} {...optionProps} />
+								))}
+							</DropMenu.OptionGroup>
+						);
```

Comment:
Do we need the curly braces and the explicit `return`? (Especially since you're already doing this in the inner `map` call :D.)

```
_.map(optionGroups, (optionGroupProps, optionGroupIndex) => (
  <DropMenu.OptionGroup key={'SingleSelectOptionGroup' + optionGroupIndex} {...optionGroupProps}>
    {optionGroupProps.children}
    {_.map(_.get(optionGroupDataLookup, optionGroupIndex), ({ optionProps, optionIndex }) => (
      <DropMenu.Option key={'SingleSelectOption' + optionIndex} {...optionProps} />
    ))}
  </DropMenu.OptionGroup>
))
```